studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddress.tsx

```diff
@@ -55,153 +38,173 @@ const BillingAddress = () => {
     'stripe.customer'
   )
 
-  const onValidate = (values: any) => {
+  const initialValues = { city, country, line1, line2, postal_code, state }
+
+  const validate = (values: any) => {
     const errors = {} as any
     if (
       (values.line1 || values.line2 || values.postal_code || values.state || values.city) &&
       !values.country
     ) {
       errors['country'] = 'Please select a country'
     }
-    if (isEqual(values, initialValues)) {
-      setIsDirty(false)
-    } else {
-      setIsDirty(true)
-    }
     return errors
   }
 
-  const onSubmit = async (values: any) => {
+  const onSubmit = async (values: any, { resetForm }: any) => {
     if (!slug) return console.error('Slug is required')
-    updateCustomer({ slug, address: values })
+
+    try {
+      await updateCustomerProfile({ slug, address: values })
```

Comment:
nit: likewise could we stick to the previous way of `mutate` instead of `mutateAsync`, then no need for unnecessary try catch here

Reply:
I haven't actually touched this component - it only shows up as diff as I removed the old BillingSettings.tsx and renamed the BillingSettingsV2 to BillingSettings - GH wrongfully shows it as diff between these files, while it was really just a removal.

Let's do these changes separately